src/main/groovy/liquibase/harness/base/BaseCompatibilityTest.groovy

```diff
@@ -0,0 +1,225 @@
+        shouldRunChangeSet = expectedResultSet != null
+        assert shouldRunChangeSet: "No expectedResultSet for ${testInput.change} against " +
+                "${testInput.database.shortName} ${testInput.database.databaseMajorVersion}." +
+                "${testInput.database.databaseMinorVersion}"
+
+        and: "check database under test is online"
+        def connection = testInput.database.getConnection()
+        shouldRunChangeSet = connection instanceof JdbcConnection
+        assert shouldRunChangeSet: "Database ${testInput.databaseName} ${testInput.version} is offline!"
+
+        and: "execute Liquibase validate command to ensure a chagelog is valid"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("validate", argsMap) //TODO: Doesn't work for sql-formatted changelogs. Add issue link
+
+        when: "execute XML, YAML, SQL and JSON formatted changelogs using liquibase update command"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("update", argsMap)
+
+        and: "execute Liquibase tag command. Tagging last row of DATABASECHANGELOG table"
+        argsMap.remove("changeLogFile")
+        argsMap.put("tag", "test_tag")
+        executeCommandScope("tag", argsMap)
+
+        and: "execute Liquibase history command"
+        executeCommandScope("history", argsMap)
+
+        and: "execute Liquibase status command"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("status", argsMap)
+
+        then: "execute metadata checking sql, obtain result set, compare it to expected result set"
+        Connection newConnection
+        JSONArray generatedResultSetArray
+        try {
+            ResultSet resultSet
+            if (connection.isClosed()) {
+                newConnection = DriverManager.getConnection(testInput.url, testInput.username, testInput.password)
```

Comment:
Can we simplify it by removing duplicated code out of if-else? 
`connection` obj isn't used later, so no need to create separate `newConnection`
```
if (connection.isClosed()) {
                connection = DriverManager.getConnection(testInput.url, testInput.username, testInput.password)
            }
```

Reply:
++